tools/clang/test/CodeGenHLSL/structInBuffer.hlsl

```diff
@@ -0,0 +1,25 @@
+// RUN: %dxc -E main  -T cs_6_0 %s
+
+struct Foo {
+    int a;
+    int b;
+    int c;
+    int d;
+};
+
+Buffer<Foo> inputs : register(t1);
+RWBuffer< int > g_Intensities : register(u1);
+
+groupshared Foo sharedData;
+
+#ifdef DX12
```

Comment:
We should either compile with the root signature unconditionally or just delete it.